zcash_primitives/src/block.rs

```diff
@@ -0,0 +1,209 @@
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use hex;
+use std::fmt;
+use std::io::{self, Read, Write};
+use std::ops::Deref;
+
+use serialize::Vector;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct BlockHash(pub [u8; 32]);
+
+impl fmt::Display for BlockHash {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        let mut data = self.0.to_vec();
```

Comment:
Instead, you could initialize a local clone of the `self.0: [u8; 32]` and then `[T]::reverse()` on that, rather than allocating.

(Not blocking.)